ffs/types.go

```diff
@@ -460,6 +461,8 @@ type FilInfo struct {
 type FilStorage struct {
 	// ProposalCid of the deal.
 	ProposalCid cid.Cid
+	// Status is the current status of the Filecoin storage deal
+	Status storagemarket.StorageDealStatus
```

Comment:
This struct previously represented the final storage state for a deal proposal. I added `Status` to it so we can also use it to represent incremental state while the deal is progressing. Other option is to create a separate stuct for that, but they will be very similar.

Reply:
Cool. This addition might leave previous stored things in CidInfo with an "empty/default" `Status` value.
I think that shouldn't be a problem despite some _weird_ output in some APIs, right?

Reply:
Yea, I was willing to just deal with those default values, but now that I'm switching away from `Status storagemarket.StorageDealStatus` to the `StateID int` and `StateName string` values, I don't like the default of `0` there which is misleading for sure. I think I'll look more closely at a custom object for these updates. What's cool is that some of the values that start off as empty actually do get values as the deal is unfolding. I'll make sure this custom object represents those.